django/views/i18n.py

```diff
@@ -284,6 +287,22 @@ def get_javascript_catalog(locale, domain, packages):
     return catalog, plural
 
 
+def _get_locale(request):
+    if request.GET and LANGUAGE_QUERY_PARAMETER in request.GET:
```

Comment:
is `if request.GET` needed? In the "null" case, I'd expect it to be an empty QueryDict in which cause "in" checking should still work fine. I'd write it as:
```
language = request.GET.get(LANGUAGE_QUERY_PARAMETER)
if language and check_for_language(language):
   ...
```